.github/workflows/helm.yml

```diff
@@ -82,6 +92,20 @@ jobs:
         run: echo CHART_VERSION=$(yq eval .version ${{ needs.set-path.outputs.charts_path }}/Chart.yaml) >> $GITHUB_OUTPUT
         id: get-chart-version
 
+      - name: Check alpha version
+        if: ${{ inputs.force-publish }}
+        run: |
+          VERSION_PATTERN="([0-9]+\.){2}[0-9]+(-alpha${{ inputs.pr-number }}.[0-9])"
+          VERSION="${{ steps.get-chart-version.outputs.CHART_VERSION }}"
+
+          if [[ ${VERSION} =~ ${VERSION_PATTERN} ]]; then
+            echo "Alpha verion matches the pattern"
+          else
+            echo "Version ${VERSION} does not match the pattern."
+            echo "Alpha version should be of shape X.Y.Z-alpha{PR-NUMBER}.N"
+            exit 1
+          fi
+
       - name: Check version
```

Comment:
I think the Check version job will fail in if you are re-publishing a chart based on the same branch/PR won't it?

Reply:
You'll have to iterate on the N if you want to publish several alpha from the same PR

Reply:
Even for flux reconciliation I think it's better to iterate on alpha versions too, don't you think ?

And concerning the merge we have one step that we want to run every time, and the other only to check the pattern in the case of an alpha version. I didn't found an easier way to do it but I can try to factorise it 👍

Reply:
Not very relevant if we don't want to override versions I think !

Reply:
But can we override versions ? It would imply to bypass the check "already exists" for alpha versions which seems a bit dangerous no ?

Reply:
Original purpose was enforcing by design a unique alpha version based on a PR number. Let's resolve this thread as it has been discussed off and it is not the way we want to pursue.